components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -4500,27 +4509,15 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
 
     private async guardCostCenterAccess(
         ctx: TraceContext,
-        userId: string,
         attributionId: AttributionId,
         operation: ResourceAccessOp,
-    ): Promise<void> {
-        traceAPIParams(ctx, { userId, attributionId });
-
-        let owner: GuardedCostCenter["owner"];
-        switch (attributionId.kind) {
-            case "team":
-                const team = await this.teamDB.findTeamById(attributionId.teamId);
-                if (!team) {
-                    throw new ResponseError(ErrorCodes.NOT_FOUND, "Team not found");
-                }
-                const members = await this.teamDB.findMembersByTeam(team.id);
-                owner = { kind: "team", team, members };
-                break;
-            default:
-                throw new ResponseError(ErrorCodes.BAD_REQUEST, "Invalid attributionId");
+        fineGrainedOp: OrganizationPermission | "not_implemented",
+    ): Promise<{ team: Team; members: TeamMemberInfo[] }> {
+        if (attributionId.kind !== "team") {
+            throw new ResponseError(ErrorCodes.BAD_REQUEST, "Invalid attributionId");
         }
 
-        await this.guardAccess({ kind: "costCenter", /*subject: costCenter,*/ owner }, operation);
+        return await this.guardTeamOperation(attributionId.teamId, operation, fineGrainedOp);
     }
```

Comment:
This is probably the biggest change in this PR. Because we don't have `user` level attribution, we can throw away most of it and the function mostly acts as an Attribution ID check.

A further cleanup would be to actually remove it entirely and use the `guardTeamOperation` instead, but that makes the current PR diff a bit too confusing, so will do in a follow-up.